search/views.py

```diff
@@ -37,14 +37,48 @@ def error404(request):
 def error500(request):
     return render(request, 'cosmos/error/HTTP500.html')
 
+# Generate all combinations of a search literal
+def generate_perm(query):
+
+    # Use bitmasking to generate all cases where the first letter of a word in the query is capitalized.
+    binary = []
+    final = []
+
+    n = len(query.split('_'))
+    for i in range(pow(2,n)):
+        b = bin(i)[2:].zfill(n)
+        binary.append(b)
+
+
+    # Create a query for the bitmask.
+
+    for b in binary:
+        perms = []
+        count = 0
+        for word in query.split('_'):
+            if b[count] == '0':
+                perms.append(word[0].lower() + word[1:]) 
+            else:
+                perms.append(word[0].upper() + word[1:] )
+            count = count+1
+        p='_'.join(perms)
+        final.append(p)
```

Comment:
In my opinion, we should sort combinations by length and/or else.
In general, one want get the most matched result with the tags she/he typed.